src/core/geocms/geonode/qgsgeonoderequest.cpp

```diff
@@ -269,111 +269,108 @@ QList<QgsGeoNodeRequest::ServiceLayerDetail> QgsGeoNodeRequest::parseLayers( con
   const QJsonObject jsonObject = jsonDocument.object();
   const QVariantMap jsonVariantMap = jsonObject.toVariantMap();
   const QVariantList layerList = jsonVariantMap.value( QStringLiteral( "objects" ) ).toList();
-  qint16 majorVersion;
-  qint16 minorVersion;
-  if ( jsonVariantMap.contains( QStringLiteral( "geonode_version" ) ) )
+
+  QString wmsURLFormat, wfsURLFormat, xyzURLFormat;
+
+  for ( const QVariant &layer : qgis::as_const( layerList ) )
   {
-    QRegularExpression re( "((\\d+)(\\.\\d+))" );
-    QRegularExpressionMatch match = re.match( jsonVariantMap.value( QStringLiteral( "geonode_version" ) ).toString() );
-    if ( match.hasMatch() )
-    {
-      const QStringList geonodeVersionSplit = match.captured( 0 ).split( '.' );
-      majorVersion = geonodeVersionSplit.at( 0 ).toInt();
-      minorVersion = geonodeVersionSplit.at( 1 ).toInt();
-    }
-    else
+    QgsGeoNodeRequest::ServiceLayerDetail layerStruct;
+    const QVariantMap layerMap = layer.toMap();
+    QVariantList layerLinks = layerMap.value( QStringLiteral( "links" ) ).toList();
+
+    if ( layerMap.value( QStringLiteral( "typename" ) ).toString().isEmpty() )
     {
-      return layers;
+      const QStringList splitURL = layerMap.value( QStringLiteral( "detail_url" ) ).toString().split( '/' );
+      layerStruct.typeName = splitURL.at( splitURL.length() - 1 );
```

Comment:
Please add a check that  "splitURL.length() - 1" is a valid index or we might have a crash.